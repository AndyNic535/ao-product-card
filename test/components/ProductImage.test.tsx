import React from "react";
import render from "react-test-renderer";
import { ProductImg, ProductCard } from '../../src/components';
import { product2 } from '../data/product';


describe( 'Pruebas del componente <ProductImg />', () => {

    test( 'Debe de mostrar el componente correctamente con la imagen personalizada.', () => {

        const wrapper = render.create(

            <ProductImg img="https://pruebaTest.jpg" />

        );

        // console.log( wrapper.toJSON() );

        expect( wrapper.toJSON() ).toMatchSnapshot();

    });

    test( 'Debe de mostrar el componente con la imagen del producto.', () => {

        const wrapper = render.create(

            <ProductCard product={ product2 }>
                {
                    () => ( 

                        <ProductImg /> 
                        
                    )
                }
            </ProductCard>

        );

        // console.log( wrapper.toJSON() );

        expect( wrapper.toJSON() ).toMatchSnapshot();

    });

});