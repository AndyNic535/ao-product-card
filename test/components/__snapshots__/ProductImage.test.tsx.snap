// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas del componente <ProductImg /> Debe de mostrar el componente con la imagen del producto. 1`] = `
<div
  className="productCard undefined"
>
  <img
    alt="Nombre Imagen"
    className="productImg undefined"
    src="./coffee-mug.png"
  />
</div>
`;

exports[`Pruebas del componente <ProductImg /> Debe de mostrar el componente correctamente con la imagen personalizada. 1`] = `
<img
  alt="Nombre Imagen"
  className="productImg undefined"
  src="https://pruebaTest.jpg"
/>
`;
